directives/abstract_exercise.py

```diff
@@ -1,9 +1,25 @@
 import itertools
+from urllib.parse import urlparse
 
 from docutils.parsers.rst import Directive, directives
 from sphinx.errors import SphinxError
 
 
+def file_mapping(argument):
+    """
+    Converts a comma-separated list of key:value pairs into a dict.
+    (Directive option conversion function.)
+    """
+    args = [arg.split(":") for arg in argument.split(',')]
+    for arg in args:
+        if len(arg) > 2 or len(arg) == 0:
+            raise SphinxError(
+                "File mapping must be in the format '<mapping 1>,<mapping 2>,...',"
+                " where each <mapping X> is either '<path on grader>:<path in repository>'"
+                " or just '<path>' if no renaming is neccessary.")
```

Comment:
I had though `<>` is a common way to represent placeholders. I guess not. `<mapping 1>` should be fine because `<mapping X> is either ...` is specified right after, right? I understand that it might be confusing for the paths, though. I also added an example to remove any ambiquity.